tests/MongoDB.Driver.Tests/OcspIntegrationTests.cs

```diff
@@ -0,0 +1,137 @@
+/* Copyright 2020–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+
+using System;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.Core.Configuration;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using MongoDB.Driver.TestHelpers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace MongoDB.Driver.Tests
+{
+    public class OcspIntegrationTests
+    {
+        private static readonly string _ocspOutcomeEnvironmentVariableName = "OCSP_TLS_SHOULD_SUCCEED";
+        private readonly ITestOutputHelper _output;
+
+        public OcspIntegrationTests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        /*
+         * This test must be run as its own task in EG.
+         * When testing locally, this test should not run alongside other integration tests.
+         * Clearing local OCSP caches between test runs is also required when testing locally.
+         * Clearing the OCSP caches may involve clearing out the caches in ~\.dotnet folder
+         * and $DOTNET_CLI_HOME/.dotnet, as well as running `dotnet clean` prior to running the tests.
+         * See spec for OCSP OS-level cache manipulation commands.
+         *   https://github.com/mongodb/specifications/blob/master/source/ocsp-support/ocsp-support.rst#os-level-ocsp-cache-manipulation
+         * When testing on Windows, the certificate should be added to the trust store prior to each run in order to
+         * reduce the chances of Windows pruning the certificate from the trust store prior to the test running.
+         */
+        [SkippableFact]
+        public void MongoClientShouldRespectCertificateStatusAndTlsInsecure()
+        {
+            /* We cannot call RequireServer.Check() because this would result in a connection being made to the mongod
+             * and that connection will not succeed if we're testing the revoked certificate case. */
+
+            // Uncommenting the line below can help if with local testing
+            // SetupLocalTestingEnvironmentVariables(mockResponderValidatesAllCertificates: false);
```

Comment:
If anyone would like to run this test locally, I can offer some assistance although it's probably faster at this point to just run the test in EG